File: IT/Variabili, operatori, strutture di controllo/Esercizi/Java/Esercizio22.java
/* file Esercizio22.java */

public class Esercizio22 {
	public static void main(String [] args) {
		String s = System.console().readLine("Scrivi un numero: ");

		if (!s.matches("[\\d]+")) {
			System.out.println("Numero non valido!");
			return;
		}

		int n = Integer.parseInt(s);

		int q = n * n;

		System.out.println(n + " al quadrato = " + q);
	}
}

File: IT/Variabili, operatori, strutture di controllo/Esercizi/C++/esercizio22.cpp
/* 
file esercizio22.cpp
compilare con: g++ -std=c++11 esercizio22.cpp
*/

#include <iostream>

using std::cin;
using std::cout;
using std::endl;
using std::string;

int main() 
{
	cout << "Scrivi un numero: ";

	string s;
	getline(cin, s);

	if (s.find_first_not_of("0123456789") != string::npos) {
		cout << "Numero non valido!" << endl;
		return 0;
	}

	int n = stoi(s);

	int q = n * n;

	cout << n << " al quadrato = " << q << endl;
}

